system_tests/timeboost_test.go

```diff
@@ -40,6 +40,114 @@ import (
+	Require(t, err)
+	err = bobExpressLaneClient.SendTransaction(ctx, duringRoundTransferTx)
+	Require(t, err)
+
+	// Alice and Bob submit bids and Alice wins
+	t.Logf("Alice and Bob now submitting their bids at %v", time.Now())
+	aliceBid, err := aliceBidderClient.Bid(ctx, big.NewInt(2), aliceOpts.From)
+	Require(t, err)
+	bobBid, err := bobBidderClient.Bid(ctx, big.NewInt(1), bobOpts.From)
+	Require(t, err)
+	t.Logf("Alice bid %+v", aliceBid)
+	t.Logf("Bob bid %+v", bobBid)
+
+	// Subscribe to auction resolutions and wait for Alice to win the auction.
+	winner, winnerRound := awaitAuctionResolved(t, ctx, seqClient, auctionContract)
+	fmt.Println("winner is ", winner, winnerRound)
+
+	// Verify Alice owns the express lane this round
+	if winner != aliceOpts.From {
+		t.Fatal("Alice should have won the express lane auction")
+	}
+	if winnerRound != currRound+1 {
+		t.Fatalf("winner round mismatch. Want: %d, Got: %d", currRound+1, winnerRound)
+	}
+	t.Log("Alice won the express lane auction for upcoming round, now try to transfer control before the next round begins...")
+
+	// Alice now transfers control to bob before her round begins
+	currRound = timeboost.CurrentRound(time.Unix(info.OffsetTimestamp, 0), roundDuration)
+	if currRound >= winnerRound {
+		t.Fatal("next round already began, try running the test again")
+	}
+
+	beforeRoundTransferTx, err := auctionContract.ExpressLaneAuctionTransactor.TransferExpressLaneController(&aliceOpts, winnerRound, seqInfo.Accounts["Bob"].Address)
+	Require(t, err)
+	err = aliceExpressLaneClient.SendTransaction(ctx, beforeRoundTransferTx)
+	Require(t, err)
+
+	setExpressLaneIterator, err := auctionContract.FilterSetExpressLaneController(&bind.FilterOpts{Context: ctx}, nil, nil, nil)
+	Require(t, err)
+	verifyControllerChange := func(round uint64, prev, new common.Address) {
+		setExpressLaneIterator.Next()
+		if setExpressLaneIterator.Event.Round != round {
+			t.Fatalf("unexpected round number. Want: %d, Got: %d", round, setExpressLaneIterator.Event.Round)
+		}
+		if setExpressLaneIterator.Event.PreviousExpressLaneController != prev {
+			t.Fatalf("unexpected previous express lane controller. Want: %v, Got: %v", prev, setExpressLaneIterator.Event.PreviousExpressLaneController)
+		}
+		if setExpressLaneIterator.Event.NewExpressLaneController != new {
+			t.Fatalf("unexpected new express lane controller. Want: %v, Got: %v", new, setExpressLaneIterator.Event.NewExpressLaneController)
+		}
+	}
+	// Verify during round control change
+	verifyControllerChange(currRound, common.Address{}, bobOpts.From) // Bob wins auction
+	verifyControllerChange(currRound, bobOpts.From, aliceOpts.From)   // Bob transfers control to Alice
+	// Verify before round control change
+	verifyControllerChange(winnerRound, common.Address{}, aliceOpts.From) // Alice wins auction
+	verifyControllerChange(winnerRound, aliceOpts.From, bobOpts.From)     // Alice transfers control to Bob before the round begins
+}
```

Comment:
Can you test that Bob's txs are sequenced before Alice's here? Maybe factor that out into its own function since this check is needed in a few places.